detectors/node/opentelemetry-resource-detector-container/src/detectors/ContainerDetector.ts

```diff
@@ -100,6 +107,43 @@ export class ContainerDetector implements Detector {
     return containerIdStr || '';
   }
 
+  private _isInKubernetesEnvironment(): boolean {
+    return process.env.KUBERNETES_SERVICE_HOST !== undefined;
+  }
+
+  private async _getContainerIdK8(): Promise<string> {
+    const kubeconfig = new KubeConfig();
+    kubeconfig.loadFromDefault();
+
+    const api = kubeconfig.makeApiClient(CoreV1Api);
+    const namespace: string = process.env.NAMESPACE || 'default';
+    const containerName: string | undefined = process.env.CONTAINER_NAME;
+    if (!containerName) {
+      throw new Error('Container name not specified in environment');
+    }
+
+    const response = await api.listNamespacePod(namespace);
```

Comment:
Is this listing all pods in the namespace?  What if there are multiple pods in the namespace with the same container name (as would happen with a k8s depoyment)?  Will each detect a random container ID?

Listing all pods in a namespace is also much more expensive than getting a single pod.

Reply:
Hello @dashpole 
I believe this statement is true where it will list all pods.

May be I can provide  a condition where it will look for a specific pod defined as environment variable.